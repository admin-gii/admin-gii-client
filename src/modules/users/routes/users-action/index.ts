export * from './users-action.component';
