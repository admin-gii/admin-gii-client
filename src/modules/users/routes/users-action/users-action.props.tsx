import { OptionProps } from '@blueprintjs/core';
import { CreateUserForm } from '@users/model/user.model';
import { parseQueryToString } from '@utils/get-query';
import { useEffect, useState } from 'react';
import { useForm } from 'react-hook-form';
import { useHistory } from 'react-router-dom';

/**
 * <UsersAction /> props
 */
export type UsersActionProps = {}

const initialValues: CreateUserForm = {
  email: '',
  phone: '',
  first_name: '',
  last_name: '',
  role: '',
  status: true
};

const roles: OptionProps[] = [
  { label: 'Admin', value: 'admin' },
  { label: 'User', value: 'user' }
];

export const useUsersActionProps = (_?: UsersActionProps) => {
  const history = useHistory();
  const [updatedItem, setUpdatedItem] = useState(null);
  const [isCreate, setIsCreate] = useState(true);
  const form = useForm({
    defaultValues: initialValues,
    mode: 'onBlur'
  });

  const onSubmit = (values: CreateUserForm) => {
    console.log({ values });
  };

  useEffect(() => {
    const searchObject = parseQueryToString(history.location.search);
    if (searchObject.add === 'new') {
      setIsCreate(true);
    } else if (searchObject.edit) {
      setUpdatedItem(searchObject.edit);
    }
  }, [history.location.search]);

  return {
    updatedItem,
    isCreate,
    onSubmit,
    roles,
    form
  };
};
