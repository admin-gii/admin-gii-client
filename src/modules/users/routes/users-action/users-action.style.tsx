import styled from 'styled-components'

export const UsersActionContainer = styled.div`

`