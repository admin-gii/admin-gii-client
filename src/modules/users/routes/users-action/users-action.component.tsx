import { hoc } from '@utils/hoc';
import { useUsersActionProps } from './users-action.props';
import { UsersActionContainer } from './users-action.style';
import { Button, H1 } from '@blueprintjs/core';
import { Error } from '@components/error';
import { InputField } from '@components/form/fields/input-field';
import { Space } from '@styles/space';
import { SwitchField } from '@components/form/fields/switch-field';
import { SuggestField } from '@components/form/fields/suggest-field';
/**
 * <UsersAction />
 */
export const UsersAction = hoc(useUsersActionProps,
  ({ updatedItem, isCreate, onSubmit, roles, form }) => {
    if (!updatedItem && !isCreate)
      return <Error status='404' text='Bunday sahifa topilmadi' />;
    return (
  <UsersActionContainer>
        <H1>Yangi foydalanuvchi qo'shish</H1>
        <Space height='2rem' />
        <form className='w-half' onSubmit={form.handleSubmit(onSubmit)}>
        <InputField
            control={form.control}
            type='email'
            name='email'
            label='Email'
            rules={{ required: 'Emailingizni kiriting!' }}
          />
          <InputField
            control={form.control}
            type='phone'
            name='phone'
            label='Telefon'
            rules={{ required: 'Telefon raqamingizni kiriting!' }}
          />
          <InputField
            control={form.control}
            type='text'
            name='first_name'
            label='Ism'
            rules={{ required: 'Ismingizni kiriting!' }}
          />
          <InputField
            control={form.control}
            type='text'
            name='last_name'
            label='Familiya'
            rules={{ required: 'Familiyangizni kiriting!' }}
          />
          <SwitchField name='status' control={form.control} label='Holati' />
          <SuggestField
            control={form.control}
            name='role'
            label='Rollar'
            items={roles}
          />
          <Space height='1rem' />
          <Button
            type='submit'
            intent='success'
            icon='cloud-upload'
            text='Saqlash'
          />
        </form>
  </UsersActionContainer>
    );
  }
);