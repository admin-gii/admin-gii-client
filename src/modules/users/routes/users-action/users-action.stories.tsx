import { ComponentStory, ComponentMeta } from '@storybook/react';
import { UsersAction } from './users-action.component';
import '@styles/index.css';

export default {
  title: 'Common/UsersAction',
  component: UsersAction,
} as ComponentMeta<typeof UsersAction>

const Template: ComponentStory<typeof UsersAction> = (args) => <UsersAction {...args} />;

export const UsersActionInitial = Template.bind({});
UsersActionInitial.args = {
}