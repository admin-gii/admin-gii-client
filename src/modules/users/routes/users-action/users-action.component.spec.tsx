import { customRender } from '@test-utils';
import { UsersAction } from './users-action.component';

describe('UsersAction', () => {
  test('Should render UsersAction component', async () => {
    const { asFragment } = customRender(
      <UsersAction />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});