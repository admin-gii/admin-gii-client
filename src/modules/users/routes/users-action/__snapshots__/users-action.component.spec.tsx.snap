// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersAction Should render UsersAction component 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 2rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 1rem;
}

<div
    class=""
  >
    <h1
      class="bp4-heading"
    >
      Yangi foydalanuvchi qo'shish
    </h1>
    <div
      class="c0"
      height="2rem"
    />
    <form
      class="w-half"
    >
      <div
        class="bp4-form-group"
      >
        <label
          class="bp4-label"
          for="email-input"
        >
          Email 
          <span
            class="bp4-text-muted"
          >
            (majburiy)
          </span>
        </label>
        <div
          class="bp4-form-content"
        >
          <div
            class="bp4-input-group"
          >
            <input
              class="bp4-input"
              id="email-input"
              name="email"
              type="email"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="bp4-form-group"
      >
        <label
          class="bp4-label"
          for="phone-input"
        >
          Telefon 
          <span
            class="bp4-text-muted"
          >
            (majburiy)
          </span>
        </label>
        <div
          class="bp4-form-content"
        >
          <div
            class="bp4-input-group"
          >
            <input
              class="bp4-input"
              id="phone-input"
              name="phone"
              type="phone"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="bp4-form-group"
      >
        <label
          class="bp4-label"
          for="first_name-input"
        >
          Ism 
          <span
            class="bp4-text-muted"
          >
            (majburiy)
          </span>
        </label>
        <div
          class="bp4-form-content"
        >
          <div
            class="bp4-input-group"
          >
            <input
              class="bp4-input"
              id="first_name-input"
              name="first_name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="bp4-form-group"
      >
        <label
          class="bp4-label"
          for="last_name-input"
        >
          Familiya 
          <span
            class="bp4-text-muted"
          >
            (majburiy)
          </span>
        </label>
        <div
          class="bp4-form-content"
        >
          <div
            class="bp4-input-group"
          >
            <input
              class="bp4-input"
              id="last_name-input"
              name="last_name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="bp4-form-group"
      >
        <div
          class="bp4-form-content"
        >
          <label
            class="bp4-control bp4-switch bp4-large"
          >
            <input
              checked=""
              id="status-input"
              type="checkbox"
            />
            <span
              class="bp4-control-indicator"
            />
            Holati
          </label>
        </div>
      </div>
      <div
        class="bp4-form-group"
      >
        <label
          class="bp4-label"
          for="role-input"
        >
          Rollar 
          <span
            class="bp4-text-muted"
          />
        </label>
        <div
          class="bp4-form-content"
        >
          <div
            class="bp4-popover-wrapper bp4-fill"
          >
            <div
              aria-haspopup="true"
              class="bp4-popover-target"
            >
              <div
                class="bp4-input-group bp4-fill"
              >
                <input
                  autocomplete="off"
                  class="bp4-input"
                  placeholder="Search..."
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c1"
        height="1rem"
      />
      <button
        class="bp4-button bp4-intent-success"
        type="submit"
      >
        <span
          aria-hidden="true"
          class="bp4-icon bp4-icon-cloud-upload"
          icon="cloud-upload"
          tabindex="-1"
        >
          <svg
            data-icon="cloud-upload"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M8.71 7.29C8.53 7.11 8.28 7 8 7s-.53.11-.71.29l-3 3a1.003 1.003 0 001.42 1.42L7 10.41V15c0 .55.45 1 1 1s1-.45 1-1v-4.59l1.29 1.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71l-3-3zM12 4c-.03 0-.07 0-.1.01A5 5 0 002 5c0 .11.01.22.02.33a3.495 3.495 0 00.07 6.37c-.05-.23-.09-.46-.09-.7 0-.83.34-1.58.88-2.12l3-3a2.993 2.993 0 014.24 0l3 3c.54.54.88 1.29.88 2.12 0 .16-.02.32-.05.47C15.17 10.78 16 9.5 16 8c0-2.21-1.79-4-4-4z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="bp4-button-text"
        >
          Saqlash
        </span>
      </button>
    </form>
  </div>
</DocumentFragment>
`;
